File: C++SrcFiles/ShootScript.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
#include <algorithm>

using namespace std;

class ShootScript : public MonoBehaviour
{
private:
    GameObject* arrowPrefab;
    Transform* arrowSpawn;
    float attackTimer;

public:
    ShootScript()
    {
        arrowPrefab = Resources::Load<GameObject>("Prefabs/Arrow");
        arrowSpawn = transform->Find("ArrowSpawn");
        attackTimer = 0.0f;
    }

    void Update()
    {
        attackTimer += Time.deltaTime;

        if (attackTimer >= 2.0f) {
            attackTimer = 0.0f;
            Shoot();
        }
    }

    void Shoot()
    {
        Instantiate(arrowPrefab, arrowSpawn->position, Quaternion::identity);
    }
};

File: C#SrcFiles/ShootScript.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ShootScript : MonoBehaviour
{
    // the prefab of the reaperattack png;
    public GameObject Arrow;
    //position of the attack;
    public Transform Arrowspawn;

    // Intervals between attacks;
    private float attacktimer;
    void Update()
    {
        // counting till 3 seconds and then shooting the reaper bullet.
        attacktimer += Time.deltaTime;
        if (attacktimer > 2)
        {
            attacktimer = 0;
            shoot();
        }
    }


    private void shoot()
    {
        Instantiate(Arrow, Arrowspawn.transform.position, Quaternion.identity);
    }
}
